modules/pathtracing/include/inviwo/pathtracing/processors/volumepathtracer.h

```diff
@@ -0,0 +1,115 @@
+ *
+ *********************************************************************************/
+
+#pragma once
+
+#include <chrono>
+
+#include <inviwo/pathtracing/pathtracingmoduledefine.h>
+#include <inviwo/core/common/inviwo.h>
+#include <inviwo/core/datastructures/light/baselightsource.h>
+#include <inviwo/core/ports/bufferport.h>
+#include <inviwo/core/ports/imageport.h>
+#include <inviwo/core/processors/processor.h>
+#include <inviwo/core/util/timer.h>
+
+#include <inviwo/core/properties/raycastingproperty.h>          // for RaycastingProperty
+#include <inviwo/core/properties/cameraproperty.h>              // for CameraProperty
+#include <inviwo/core/properties/optionproperty.h>              // for OptionPropertyInt
+#include <inviwo/core/properties/volumeindicatorproperty.h>     // for VolumeIndicatorProperty
+#include <inviwo/core/properties/transferfunctionproperty.h>    // for TransferFunctionProperty
+#include <inviwo/core/datastructures/light/baselightsource.h>   // for Lights
+#include <inviwo/core/ports/bufferport.h>                       // for Lights
+#include <inviwo/core/properties/simplelightingproperty.h>      // for SimpleLightingProperty
+#include <inviwo/core/properties/buttonproperty.h>              // for IterativeRendering
+#include <inviwo/core/util/timer.h>                             // for IterativeRendering
+
+// outside of include/inviwo
+#include <../modules/opengl/include/modules/opengl/volume/volumegl.h>
+#include <../modules/opengl/include/modules/opengl/volume/volumeutils.h>
+#include <../modules/opengl/include/modules/opengl/shader/shader.h>
+#include <../modules/opengl/include/modules/opengl/image/layergl.h>
+#include <../modules/opengl/include/modules/opengl/buffer/buffergl.h>
+
+namespace inviwo {
+
+class IVW_MODULE_PATHTRACING_API VolumePathTracer : public Processor {
+public:
+    VolumePathTracer();
+    ~VolumePathTracer() = default;
+
+    virtual void initializeResources() override {
+        invalidateProgressiveRendering();
+    }
+    virtual void process() override;
+
+    virtual const ProcessorInfo getProcessorInfo() const override;
+    static const ProcessorInfo processorInfo_;
+
+protected:
+    void numSamplesChanged(); 
+    void updateLightSources();  
+    void invalidateProgressiveRendering(); 
+    void evaluateProgressiveRefinement(); 
+    void progressiveRefinementChanged(); 
+    //void phaseFunctionChanged(); 
+    void onTimerEvent(); 
+
+private:
+    // Ports
+    VolumeInport volumePort_;
+    ImageInport entryPort_;
+    ImageInport exitPort_; 
+    
+    ImageOutport outport_;
+
+    // Properties and Internals
+    Shader shader_;
+    OptionPropertyInt channel_;
+    RaycastingProperty raycasting_;
+    TransferFunctionProperty transferFunction_;
+
+    CameraProperty camera_;
+
+    VolumeIndicatorProperty positionIndicator_;
+
+    SimpleLightingProperty light_;
+    std::chrono::time_point<std::chrono::high_resolution_clock> time_start;
```

Comment:
We use camelCase convention with underscore for member variables
```suggestion
    std::chrono::time_point<std::chrono::high_resolution_clock> timeStart_;
```